config/distscript.pl

```diff
@@ -50,6 +50,28 @@ sub subst {
 }
 
 ###############################################################################
+# Check to see that the source tree is clean / has no local changes
+###############################################################################
+
+if (-d ".git") {
+    open(GIT_STATUS, "git status --porcelain|") ||
+        die "Can't run git status to verify that the source tree is clean";
+    my $clean = 1;
+    while (<GIT_STATUS>) {
+        chomp;
+        if ($_ =~ m/^\s*M (.+)$/) {
```

Comment:
This pattern is too strict.  It would fail to match:
```
git rebase
First, rewinding head to replay your work on top of it...
Applying: WIP demo of merge conflict
Using index info to reconstruct a base tree...
M       prov/verbs/src/fi_verbs.c
Falling back to patching base and 3-way merge...
Auto-merging prov/verbs/src/fi_verbs.c
CONFLICT (content): Merge conflict in prov/verbs/src/fi_verbs.c
Failed to merge in the changes.
Patch failed at 0001 WIP demo of merge conflict
The copy of the patch that failed is found in:
   /home/dgoodell/git/libfabric/.git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

savbu-usnic-a ~/g/libfabric git:remotes/origin/HEAD:rebase ❯❯❯ git status
rebase in progress; onto 568839f
You are currently rebasing branch 'usnic' on '568839f'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

        both modified:   prov/verbs/src/fi_verbs.c

no changes added to commit (use "git add" and/or "git commit -a")
savbu-usnic-a ~/g/libfabric git:remotes/origin/HEAD:rebase ❯❯❯ git status --porcelain
UU prov/verbs/src/fi_verbs.c
```

Here's the possible output from `git status --porcelain` (from the man page):
```
           X          Y     Meaning
           -------------------------------------------------
                     [MD]   not updated
           M        [ MD]   updated in index
           A        [ MD]   added to index
           D         [ M]   deleted from index
           R        [ MD]   renamed in index
           C        [ MD]   copied in index
           [MARC]           index and work tree matches
           [ MARC]     M    work tree changed since index
           [ MARC]     D    deleted in work tree
           -------------------------------------------------
           D           D    unmerged, both deleted
           A           U    unmerged, added by us
           U           D    unmerged, deleted by them
           U           A    unmerged, added by them
           D           U    unmerged, deleted by us
           A           A    unmerged, both added
           U           U    unmerged, both modified
           -------------------------------------------------
           ?           ?    untracked
           !           !    ignored
           -------------------------------------------------
```

So a better pattern might be:
```perl
        if ($_ =~ m/^([^?! ].|.[^?! ]) (.+)$/) {
```
If you wanted to be extra safe you could drop the `?` from that pattern so that we reduce the risk of making a tarball that depends on untracked files.